modules/zivid/point_cloud.py

```diff
@@ -5,59 +5,153 @@
 
 
 class PointCloud:
-    """A point cloud."""
+    """Point cloud with x, y, z, RGB and color laid out on a 2D grid.
 
-    def __init__(self, internal_point_cloud):
-        """Create a point cloud from an internal point cloud.
+    An instance of this class is a handle to a point cloud stored on the compute device memory.
+    Use the method copy_data to copy point cloud data from the compute device to a numpy
+    array in host memory. Several formats are available.
+    """
 
-        Args:
-            internal_point_cloud: a internal point cloud
+    class Downsampling:  # pylint: disable=too-few-public-methods
+        """Collection of valid options to PointCloud.downsample()."""
 
-        Raises:
-            TypeError: unsupported type provided for internal point cloud
+        by2x2 = "by2x2"
+        by3x3 = "by3x3"
+        by4x4 = "by4x4"
 
-        """
-        if not isinstance(internal_point_cloud, _zivid.PointCloud):
+        _valid_values = {
+            "by2x2": _zivid.PointCloud.Downsampling.by2x2,
+            "by3x3": _zivid.PointCloud.Downsampling.by3x3,
+            "by4x4": _zivid.PointCloud.Downsampling.by4x4,
+        }
+
+        @classmethod
+        def valid_values(cls):
+            """Get list of allowed values.
+
+            Returns:
+                List of strings
+            """
+            return list(cls._valid_values.keys())
+
+    def __init__(self, impl):  # noqa: D107
+        if not isinstance(impl, _zivid.PointCloud):
             raise TypeError(
                 "Unsupported type for argument internal_point_cloud. Got {}, expected {}".format(
-                    type(internal_point_cloud), type(_zivid.PointCloud)
+                    type(impl), type(_zivid.PointCloud)
                 )
             )
-        self.__impl = internal_point_cloud
+        self.__impl = impl
+
+    def copy_data(self, data_format):
+        """Copy point cloud data from GPU to numpy array.
 
-    def to_array(self):
-        """Convert point cloud to numpy array.
+        Supported data formats:
+        xyz:        ndarray(H,W,3) of float
+        xyzw:       ndarray(H,W,4) of float
+        z:          ndarray(H,W)   of float
+        rgba:       ndarray(H,W,4) of uint8
+        snr:        ndarray(H,W)   of float
+        xyzrgba:    ndarray(H,W)   of composite dtype (accessed with e.g. arr["x"])
```

Comment:
We might want to change `H` and `W` into `Height` and `Width` to make it consistent

Reply:
Done